dnssec_keyscan.go

```diff
@@ -9,11 +9,16 @@ import (
 	"math/big"
 	"strconv"
 	"strings"
+	"fmt"
 )
 
 // NewPrivateKey returns a PrivateKey by parsing the string s.
 // s should be in the same form of the BIND private key files.
 func (k *DNSKEY) NewPrivateKey(s string) (crypto.PrivateKey, error) {
+	if s == "" {
+		return nil, fmt.Errorf("Blank string specified")	
+	}
+	
 	if s[len(s)-1] != '\n' { // We need a closing newline
```

Comment:
the panic happens here, I think? It's better to not invent a completely new error msg, and just add the check to this line:
`if s == "" || s[len.... ` which should solve your problem.